src/Adapter/Presenter/Cart/CartPresenter.php

```diff
@@ -448,7 +448,8 @@ function ($voucher) {
         $discounts = array_filter($discounts, function ($discount) use ($cartRulesIds, $cart) {
             $voucherCustomerId = (int) $discount['id_customer'];
             $voucherIsRestrictedToASingleCustomer = ($voucherCustomerId !== 0);
-            if ($voucherIsRestrictedToASingleCustomer && $cart->id_customer !== $voucherCustomerId) {
+            $voucherIsEmptyCode = empty($discount['code']);
+            if ($voucherIsRestrictedToASingleCustomer && $cart->id_customer !== $voucherCustomerId && $voucherIsEmptyCode) {
```

Comment:
```suggestion
            if (($voucherIsRestrictedToASingleCustomer && $cart->id_customer !== $voucherCustomerId) || $voucherIsEmptyCode) {
```

Should it be something like this instead? :thinking: 
It looks weird to authorized `$voucherIsRestrictedToASingleCustomer = true` && `$cart->id_customer !== $voucherCustomerId` if the code is not empty :thinking:

Reply:
In fact, 
- if $voucherIsRestrictedToASingleCustomer = true && $cart->id_customer !== $voucherCustomerId && code is empty, > it's a credit note. The voutcher is automaticly apply > You don't need to display the voucher if hightlight is set on it.
- if $voucherIsRestrictedToASingleCustomer = true && $cart->id_customer !== $voucherCustomerId && but code is NOT empty, >  it's a something like a fidelity voucher, a discount. Usage is not mandatory. > You don't need to display the voucher if hightlight is set on it.

Reply:
Thanks for explanation :wink:

Reply:
Note that if
 $voucherIsRestrictedToASingleCustomer = false  && code is empty > I don't know the use case... Probably we need to display the voucher